x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRetrieverBuilder.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rank.rrf;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.retriever.RetrieverBuilder;
+import org.elasticsearch.search.retriever.RetrieverParserContext;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+public final class RRFRetrieverBuilder extends RetrieverBuilder<RRFRetrieverBuilder> {
+
+    public static final ParseField RETRIEVERS_FIELD = new ParseField("retrievers");
+    public static final ParseField WINDOW_SIZE_FIELD = new ParseField("window_size");
+    public static final ParseField RANK_CONSTANT_FIELD = new ParseField("rank_constant");
+
+    public static final ObjectParser<RRFRetrieverBuilder, RetrieverParserContext> PARSER = new ObjectParser<>(
+        RRFRankPlugin.NAME,
+        RRFRetrieverBuilder::new
+    );
+
+    static {
+        PARSER.declareObjectArray((v, l) -> v.retrieverBuilders = l, (p, c) -> {
```

Comment:
This looks like we are declaring "retrievers" field as optional for the parser, should we make it required?

Reply:
I agree with you on the required. This should probably be a constructing object parser as well like you did for knn. I will make these changes in a follow up. For now this is protected by the fact we extract to sub_searches and you need at least 2 queries there. However, there error message would be really confusing to get.

Reply:
I agree with you on the required. This should probably be a constructing object parser as well like you did for knn. I will make these changes in a follow up. For now this is protected by the fact we extract to sub_searches and you need at least 2 queries there. However, there error message would be really confusing to get.